src/CoreWCF.Primitives/src/CoreWCF/Description/OperationDescription.cs

```diff
@@ -126,7 +127,7 @@ public string Name
 
         internal KeyedByTypeCollection<IAuthorizeOperation> AuthorizeOperation { get; }
 
-        internal Lazy<ReadOnlyCollection<IAuthorizeData>> AuthorizeData { get; set; }
+        internal ConcurrentDictionary<Type, ReadOnlyCollection<IAuthorizeData>> AuthorizeData { get; set; }
```

Comment:
ConcurrentDictionary is optimized for multiple concurrent readers and writers and has some performance issues. For example it has buckets which are affinitized a thread (or multiple threads) to reduce contention, but then when reading it needs to check multiple buckets. There's a little known pattern you can do when you have a seldom write read often situation like we have here. The HashTable class has an interesting property that it can safely return results from multiple readers at once concurrent to a write, as long as you only ever have a single write happening. In other words, reads are always thread safe and a single write is thread safe with multiple reads. The failure mode with concurrent read and write to an entry being modified (value added, changed, or deleted) is the read might return a stale value, but it won't ever throw or return an incorrect value (value from different key). I was going to quickly code up a helper class for this and went looking at the implementation and today was one of those days when I learn something new :smile:. .NET has had a s[tatic method to return a wrapped implementation](https://learn.microsoft.com/en-us/dotnet/api/system.collections.hashtable.synchronized?view=net-7.0) which only locks on write since .NET Framework 1.1. You would create a new instance with `Hashtable.Synchronized(new Hashtable());`. You can either use this directly and add cases into the code where necessary, or I [quickly threw together a generic implementation](https://gist.github.com/mconnew/772049ac7e202d8e7468e5df74ecfb93) of `IDictionary<Tkey,TValue>` based on this which I did some basic testing on to validate. You can see a couple of methods still need to take a lock so avoid those if you choose to use this.